datastore/graphers/nmdb-graph-network/nmdb-graph-network.cpp

```diff
@@ -181,36 +181,41 @@ class Tool : public nmdt::AbstractGraphTool
     void addToolOptions() override
     {
       opts.addRequiredOption("layer", std::make_tuple(
-            "layer,L",
-            po::value<std::string>()->required(),
-            "Layer of the network stack to graph")
-          );
+          "layer,L",
+          po::value<std::string>()->required(),
+          "Layer of the network stack to graph")
+        );
       opts.addRequiredOption("device-id", std::make_tuple(
-            "device-id",
-            po::value<std::string>()->required(),
-            "Device ID or subnet CIDR address to use as graph's root node")
-          );
+          "device-id",
+          po::value<std::string>()->required(),
+          "Device ID or subnet CIDR address to use as graph's root node")
+        );
 
       opts.addOptionalOption("icons", std::make_tuple(
-            "icons",
-            NULL_SEMANTIC,
-            "Enable device icons in graph")
-          );
+          "icons",
+          NULL_SEMANTIC,
+          "Enable device icons in graph")
+        );
+      opts.addOptionalOption("icon-folder", std::make_tuple(
+          "icon-folder",
+          po::value<std::string>(),
+          "Set custom destination for alternate icons")
+        );
```

Comment:
This should be required and set the default value to be the current path which the tool uses.
```suggestion
      opts.addOptionalOption("icon-folder", std::make_tuple(
          "icon-folder",
          po::value<std::string>()->required()->default_value(nmfm.getConfPath()/"images"),
          "Set custom destination for alternate icons")
        );
```

My suggested edit requires you to declare `nmfm` which should be at the class level and something like `const nmcu::FileManager& nmfm {nmcu::FileManager::getInstance()};` (you can find/see this construct in other code in this repo).

Reply:
Implemented suggested edits.